### DISCUSSION TITLE
Magick++: unable to set EXIF data via C++ api

### DISCUSSION THREAD

**Discussion by whatdoineed2do** [ImageMagick/ImageMagick#7048] [Help]
### ImageMagick version

7.1.1.1-26

### Operating system, version and so on

Linux

### Description

https://www.imagemagick.org/Magick++/Image++.html suggests we can update EXIF data but I don't find this to be the case, although this could be user error.  Even using attributes as seen in MagickCore/property.c, updates don't work.

What is the reason why `attribute()` or `artifact` is not updating the values?  The output file type doesnt matter whether its jpg or png etc.  Sample code to illustrate:
``` C++
#include <iostream>
#include <array>
#include <Magick++.h>

int main(int argc, char* const argv[])
{
    try
    {
        Magick::InitializeMagick(nullptr);

        Magick::Image  img;
        if (argc == 2) {
            img.read(argv[1]);
            std::cout << "make=" << img.attribute("EXIF:Make") << "\n";  // corerctly finds attribute, if exists
        }
        else {
            img = Magick::Image("640x400", "blue");
        }
        img.attribute("EXIF:Make",  "Foo Inc");
        img.attribute("EXIF:Model", "D9000");
        std::cout << "updated make=" << img.attribute("EXIF:Make") << "\n";  // prints "Foo Inc" as per update

        constexpr std::array  output {
            "exif.png","exif.jpg"
        };
        std::for_each(output.begin(), output.end(), [&img](const auto& path) {
            Magick::Image(img).write(path);  // filesystem file contains no updates, as seen by exiftool etc
        });
    }
    catch (const std::exception& ex)
    {
        std::cerr << "failed to magick - " << ex.what() << std::endl;
    }

    return 0;
}
```

**Comment by snibgo** [comment]
> What is the reason why attribute() or artifact is not updating the values?

I haven't tried your code, but I think the values are being updated in memory. However, the values are not being written to the output files. I don't really understand why, but in general IM is not good at writing exif tags.

**Comment by whatdoineed2do** [comment]
https://github.com/ImageMagick/ImageMagick/issues/55#issuecomment-157368082

Based on this 2015 comment, exif is not writen back - at quick look in `coders/jpg.c` doesn't seem to have any specific exif process so unless a maintainer comments otherwise, I will have to accept that all _"exif" updates_ are in memory only.
